src/compilers/directives/highlight.cr

```diff
@@ -0,0 +1,36 @@
+module Mint
+  class Compiler
+    def _compile(node : Ast::Directives::Highlight) : String
+      content =
+        compile node.content
+
+      formatted =
+        Formatter.new.format(node.content, Formatter::BlockFormat::Naked)
+
+      parser = Parser.new(formatted, "source.mint")
+      parser.code_block_naked
+
+      parts =
+        SemanticTokenizer.tokenize(parser.ast)
+
+      mapped =
+        parts.map do |item|
+          case item
+          in String
+            "`#{skip { escape_for_javascript(item) }}`"
+          in Tuple(SemanticTokenizer::TokenType, String)
+            "_h('span', { className: '#{item[0].to_s.underscore}' }, [`#{skip { escape_for_javascript(item[1]) }}`])"
+          end
+        end
+
+      "[#{content}, _h(React.Fragment, {}, [#{mapped.join(",\n")}])]"
+    end
+
+    def escape_for_javascript(value : String)
```

Comment:
May be worth adding this as a method on the JS renderer? `js.string()` or similar? (I'm assuming `value` here is the rendered HTML?)

Although if this the only use might be overkill 😅

Reply:
I don't know where would it be better :shrug: It's only used in the compiler but `js` is only used in the compiler as well :smile: